src/main/java/DungeonController.java

```diff
@@ -226,9 +226,10 @@ private void removeDoor(int roomStartY, int roomStartX) {
     }
 
     public void addScore() {
-        if (gridBounds[player.getPlayerY()][player.getPlayerX()] == 2) {
-            gridBounds[player.getPlayerY()][player.getPlayerX()] = 0;
-            player.setGold(player.getGold() + golds[0].getGoldQuantity());
+        for (Gold gold : golds) {
+            if (gold.getGoldX() == player.getPlayerX() && gold.getGoldY() == player.getPlayerY()) {
+                player.setPlayerScore(player.getPlayerScore() + gold.getGoldQuantity());
```

Comment:
We should still be using player.setGold() here as the amount of gold the player is holding is different from the player's score. We would want to be able to see both.